lib/ddtrace/configuration/components.rb

```diff
@@ -106,6 +109,33 @@ def build_tracer_options(settings)
           end
         end
 
+        def build_tracer_test_mode_options(settings)
+          {}.tap do |opts|
+            # Enable partial flushing:
+            #   When in test mode, traces won't complete until tests complete.
+            #   We need to flush spans from the context before its finished so
+            #   tests can assert upon trace output, if needed.
+            opts[:partial_flush] = true
+            # NOTE: Setting threshold to 1 makes this like "span streaming"
+            #       A side effect of doing so is every span will receive both
+            #       "origin" and "sampling_priority" tags. This is expected,
+            #       as currently "origin" needed on every test span.
+            opts[:min_spans_before_partial_flush] = 1
```

Comment:
is the side effect because the root span in each flush gets assigned origin and sampling priority?

Reply:
It's a side effect because the "partial flush" always tags the top most span with origin and sampling priority. We actually do want origin on every span for tests, I don't think it matters if priority is there or not.

Reply:
No longer relevant.